core/domain/base_model_validators.py

```diff
@@ -151,6 +162,11 @@ def __init__(
                 raise utils.ValidationError(
                     'The field \'%s\' should not contain '
                     'pseudonymous IDs' % field_name)
+        for model_id in filtered_model_ids:
+            if not user_services.is_user_id_valid(model_id):
+                raise Exception(
+                    'The user id %s in the field \'%s\' is '
+                    'invalid' % (model_id, field_name))
```

Comment:
Here the usage makes since we know how the ID will look like.

Reply:
Done.